### ISSUE TITLE
REST client pom.xml has runtime dependencies

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#19835]
With alpha5-SNAPSHOT, here is an extract of the generated `pom.xml`. See https://oss.sonatype.org/content/repositories/snapshots/org/elasticsearch/client/rest/5.0.0-alpha5-SNAPSHOT/rest-5.0.0-alpha5-20160805.100826-168.pom

```xml
<dependency>
  <groupId>org.apache.httpcomponents</groupId>
  <artifactId>httpcore</artifactId>
  <version>4.4.5</version>
  <scope>runtime</scope>
</dependency>
<dependency>
  <groupId>org.apache.httpcomponents</groupId>
  <artifactId>httpasyncclient</artifactId>
  <version>4.1.2</version>
  <scope>runtime</scope>
</dependency>
```

Which basically means that you can't use OOTB the apache classes:

```java
RestClient restClient = RestClient.builder(
        new HttpHost("http", "localhost", 9200),
        new HttpHost("http", "localhost", 9201)).build();
```

This won't compile.

Note that it was working well with alpha4. I did not try a "final" released version of alpha5 though.

**Comment by dadoonet** [comment]
The workaround for now is to add:

```xml
<dependency>
    <groupId>org.apache.httpcomponents</groupId>
    <artifactId>httpcore</artifactId>
    <version>4.4.5</version>
    <scope>compile</scope>
</dependency>
<dependency>
    <groupId>org.apache.httpcomponents</groupId>
    <artifactId>httpasyncclient</artifactId>
    <version>4.1.2</version>
    <scope>compile</scope>
</dependency>
<dependency>
    <groupId>org.apache.httpcomponents</groupId>
    <artifactId>httpclient</artifactId>
    <version>4.1.2</version>
    <scope>compile</scope>
</dependency>
```

**Comment by nik9000** [comment]
Hmm - I think it'd be much more friendly to call them compile dependencies. We mostly want them to be. If the user really wants them to be runtime dependencies they can override the scope in their pom.

**Comment by javanna** [comment]
I have no idea where this runtime thing comes from, they are declared compile in our build.gradle. https://github.com/elastic/elasticsearch/blob/master/client/rest/build.gradle#L34

**Comment by nik9000** [comment]
I expect it is a pom generation thing.

**Comment by javanna** [comment]
@rjernst do you know where this may come from? let me know if I can help.

**Comment by rjernst** [comment]
This is an issue with the `maven-publish` plugin in gradle, which the unified release uses. In alpha5, we generated the poms for the client jars using the new mechanism, while the old jar poms were still built using the `maven` plugin.

I opened a PR to fix: #19876
